aten/src/ATen/native/SparseMM.cpp

```diff
@@ -0,0 +1,85 @@
+#include "ATen/ATen.h"
+#include "ATen/Dispatch.h"
+#include "ATen/NativeFunctions.h"
+
+#include <TH/TH.h>
+#include <THNN/THNN.h>
+#undef THNN_
+#include <THS/THS.h>
+
+#include "ATen/CPUFloatTensor.h"
+#include "ATen/CPUDoubleTensor.h"
+#include "ATen/SparseCPUFloatTensor.h"
+#include "ATen/SparseCPUDoubleTensor.h"
+
+namespace at {
+namespace native {
+
+// Calling into TH for sspaddmm because ATen code generation currently
+// doesn't support Sparse x Dense operations on Sparse tensors
+template <class scalar>
+void TH_sspaddmm(Tensor & result, Scalar beta, const Tensor& self,
+    Scalar alpha, const Tensor& mat1, const Tensor& mat2) {
+  runtime_error("sspaddmm NYI for passed in types\n");
```

Comment:
don't include a newline in the error message and also print the types, it would look something like:
runtime_error("... for types %s %s %s, self.type().toString(), mat1.type().toString(), mat2.type().toString().